### ISSUE TITLE
recording extension: Device runs out of memory with a relatively small programme

### ISSUE THREAD

**Issue by microbit-carlos** [microsoft/pxt-microbit#5131]
This programme records sound and then attempts to change the sampling rate with the accelerometer as it plays:

<img width="912" alt="image" src="https://github.com/microsoft/pxt-microbit/assets/29712657/5368057c-f46a-49c6-8f52-e723a5be19fe">

```ts
input.onButtonPressed(Button.A, function () {
    record.startRecording()
})
input.onButtonPressed(Button.B, function () {
    record.playAudio()
    while (record.audioStatus(record.AudioStatus.Playing)) {
        record.setSampleRate(Math.map(Math.abs(input.acceleration(Dimension.X)), 0, 1023, 5000, 15000), record.AudioSampleRateScope.Playback)
    }
})
```

If the recording is more than a second or so it will throw an out-of-memory error as soon as it is played back. And if it's shorter than that it will throw the error the second time it is played back.

> makecode.microbit.org version:  [5.1.26](https://github.com/Microsoft/pxt-microbit/releases/tag/v5.1.26)
> Microsoft MakeCode version:  [8.6.33](https://github.com/Microsoft/pxt/releases/tag/v8.6.33)
> microbit runtime version:  [v2.2.0-rc6](https://github.com/lancaster-university/microbit/releases/tag/v2.2.0-rc6)
> codal-microbit-v2 runtime version:  [v0.2.50](https://github.com/lancaster-university/codal-microbit-v2/releases/tag/v0.2.50)

**Comment by microbit-carlos** [comment]
Ah, this might be the memory leak that can be fixed when CODAL exposes the sampling rate setter:
https://github.com/microsoft/pxt-microbit/blob/64a4bf6766fe4d2b185e20f25b658254e6a6cc2a/libs/audio-recording/recording.cpp#L63-L66

**Comment by abchatra** [comment]
Tracking to ensure we address this issue by consuming correct CODAL api.

**Comment by srietkerk** [comment]
Yes, when the new API is exposed, I'll be able to fix this leak. That is where it is.

**Comment by microbit-carlos** [comment]
The accessors are implemented in the the v0.2.52 tag, so this can be updated.
